### PR TITLE
Fix and rename MultiMsMsVisualizer --> SpectraStackVisualizer

### PR DESCRIPTION
**Pull request by robinschmid**
made a module, clean up code, fix laout issues

### CODE REVIEW

**Review by omokshyna — DISMISSED** [review]
Ok, there are only small comments, otherwise everything seems good.
One question maybe. MultiMSMSTab was removed, what happened to MultiMSMSPane?

File: src/main/java/io/github/mzmine/gui/mainwindow/SimpleTab.java [hunk 1]
```diff
@@ -22,14 +22,24 @@
 import io.github.mzmine.datamodel.features.FeatureList;
 import java.util.Collection;
 import java.util.Collections;
+import javafx.scene.Node;
 import org.jetbrains.annotations.NotNull;
 
 public class SimpleTab extends MZmineTab {
 
+  public SimpleTab(String title, Node content, boolean showBinding, boolean defaultBindingState) {
```
[the remark this answers is not in the document]
Comment (robinschmid) [reply src/main/java/io/github/mzmine/gui/mainwindow/SimpleTab.java:30]: could be in the future. The difference is that the MZmineTab has so many non abstract methods that are hidden here.

File: src/main/java/io/github/mzmine/modules/io/spectraldbsubmit/view/MSMSLibrarySubmissionWindow.java [hunk 2]
```diff
@@ -417,10 +415,9 @@ private void createMetaDataPanel() {
...
     // Create labels and components for each parameter
     for (Parameter p : paramMeta.getParameters()) {
-      if (!(p instanceof UserParameter)) {
+      if (!(p instanceof UserParameter up)) {
         continue;
       }
-      UserParameter up = (UserParameter) p;
 
       Node comp = up.createEditingComponent();
       // comp.setToolTipText(up.getDescription());
```
[the remark this answers is not in the document]
Comment (robinschmid) [reply src/main/java/io/github/mzmine/modules/io/spectraldbsubmit/view/MSMSLibrarySubmissionWindow.java:423]: actually added the tooltip

File: src/main/java/io/github/mzmine/modules/io/spectraldbsubmit/view/MSMSLibrarySubmissionWindow.java [hunk 3]
```diff
@@ -449,26 +446,24 @@ private void createMetaDataPanel() {
...
+      pnMetaData.add(comp, 1, rowCounter);
       rowCounter++;
     }
   }
 
   @SuppressWarnings({"unchecked", "rawtypes"})
   protected void updateParameterSetFromComponents() {
     for (Parameter<?> p : paramSubmit.getParameters()) {
-      if (!(p instanceof UserParameter)) {
+      if (!(p instanceof UserParameter up)) {
```
[the remark this answers is not in the document]
Comment (robinschmid) [reply src/main/java/io/github/mzmine/modules/io/spectraldbsubmit/view/MSMSLibrarySubmissionWindow.java:457]: I think instanceof with pattern matching is a great way to control the flow of information. 
At least they are building more pattern matching like this now with switch instanceof etc.. and its a minor operation so no real performance hit. Especially in those cases were its not in a hotspot loop
https://stackoverflow.com/questions/103564/the-performance-impact-of-using-instanceof-in-java

File: src/main/java/io/github/mzmine/modules/visualization/spectra/spectra_stack/SpectraItemLabelGenerator.java [hunk 4]
```diff
@@ -36,15 +35,16 @@ public class SpectraItemLabelGenerator implements XYItemLabelGenerator {
    */
   public static final int POINTS_RESERVE_X = 100;
 
-  private NumberFormat mzFormat = MZmineCore.getConfiguration().getMZFormat();
+  private final NumberFormat mzFormat = MZmineCore.getConfiguration().getMZFormat();
 
-  public SpectraItemLabelGenerator() {}
+  public SpectraItemLabelGenerator() {
```
[the remark this answers is not in the document]
Comment (robinschmid) [reply src/main/java/io/github/mzmine/modules/visualization/spectra/spectra_stack/SpectraItemLabelGenerator.java]: deleted

**Review by SteffenHeu — APPROVED** [review]

**Pull request merged**